packages/gestalt/src/SideNavigation/getChildrenToArray.js

```diff
@@ -1,19 +1,16 @@
 // @flow strict
 import { Children, Fragment, type Node as ReactNode } from 'react';
+import { ALLOWED_CHILDREN_MAP } from './navigationChildrenUtils';
+import { type ReactChildArray } from '../utils/flattenChildren';
 
-const ALLOWED_CHILDREN_MAP = {
-  main: ['SideNavigation.Section', 'SideNavigation.TopItem', 'SideNavigation.Group'],
-  nested: ['SideNavigation.NestedItem', 'SideNavigation.NestedGroup'],
-};
-
+/** @deprecated */
```

Comment:
This function does 3 things:

1. Checks whether subcomponent composition is valid
2. Flattens subcomponents
3. Limits `Fragment` nesting up to 2 levels

Deprecated, because I created separate functions for the first 2 things. Removed `Fragment` nesting limitation.

It is also used in `GroupdMobile` file and I didn't want to make unrelated changes.